src/snowflake/snowpark/_internal/ast.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
```

Comment:
Can we move those AST files under a new folder `ast/` and add its ownership to IR team only.
Also, can we add this new ast folder in pyright check in `tox.ini` and make sure type hints have been added for ast code.

Reply:
Done. I also renamed `ast.py` to `batch.py` to avoid having to deal with `ast/ast.py`.

Reply:
great! let's update ownership file and enable this folder with pyright.

Reply:
* Updated ownership
* Added pyright checks

Pyright checks yield around 130 warnings and errors. A small number of them are actual minor issues. A much larger number of them has to do with protocol buffers and setting up tooling (e.g. `"Expr" is not a known member of module "snowflake.snowpark._internal.proto.generated.ast_pb2"`).

I'll file a separate task and do the pyright work in a separate pull request, if that's OK with you.

Reply:
Yes, thanks for exploring this. I see people use type-protobuf instead but it seems only works with protobuf~=5.28.3. Let's revisit it later.

Reply:
I fully support getting the type annotations right, around our protocol buffer types and in Python, by the way. I find it critical for understanding the code base at all.